janusgraph-solr/src/test/java/org/janusgraph/diskstorage/solr/SolrIndexTest.java

```diff
@@ -175,4 +186,65 @@ public void testMapKey2Field_MappingSpaces() {
         KeyInformation keyInfo = new StandardKeyInformation(Boolean.class, Cardinality.SINGLE);
         assertEquals("field•name•with•spaces_b", index.mapKey2Field("field name with spaces", keyInfo));
     }
+    
+    @Test
+    public void testUpdateListWithNoStaleValues() throws Exception {
```

Comment:
The tests are mostly identical. I would better to write a single parametrized test for stale values check. Something like:
```
public static Stream<Boolean> collectionCardinalityReplaceStaleValue() {
    return ImmutableSet.of(true, false).stream();
}

@ParameterizedTest
@MethodSource("collectionCardinalityReplaceStaleValue")
public void testCollectionCardinalityStaleValues(Boolean replaceStaleValue) throws BackendException {
    final String store = "vertex";
    final String docid = "docid";
    final String defText = "1";
    final String revisedText = "2";

    if(replaceStaleValue) {
        testingStaleValues = true;
        index = new SolrIndex(getLocalSolrTestConfig());
    }

    initialize(store);
    Multimap<String, Object> initialProps = ImmutableMultimap.of(PHONE_LIST, defText);
    add(store, docid, initialProps, true);
    clopen();

    // Sanity check
    assertEquals(docid, tx.queryStream(new IndexQuery(store, PredicateCondition.of(PHONE_LIST, Cmp.EQUAL, "1"))).findFirst().get());

    tx.add(store, docid, PHONE_LIST, revisedText, false);
    tx.commit();
    clopen();

    if(replaceStaleValue) {
        // Should no longer return old text
        assertEquals(Optional.empty(), tx.queryStream(new IndexQuery(store, PredicateCondition.of(PHONE_LIST, Cmp.EQUAL, defText))).findFirst());
    } else {
        // Without the replace-stale-values flag, the old value should still be present
        assertEquals(docid, tx.queryStream(new IndexQuery(store, PredicateCondition.of(PHONE_LIST, Cmp.EQUAL, defText))).findFirst().get());
    }

    // the new value should also be present
    assertEquals(docid, tx.queryStream(new IndexQuery(store, PredicateCondition.of(PHONE_LIST, Cmp.EQUAL, revisedText))).findFirst().get());

    testingStaleValues = false;
}
```

Reply:
Thank you for the suggestion, I updated the testing to match the example you showed above and it works as expected. I appreciate it!